packages/@aws-cdk/aws-servicecatalogappregistry/lib/private/validation.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Class to validate that inputs match requirements.
+ */
+export class InputValidator {
+  /**
+   * Validates length is between allowed min and max lengths.
+   */
+  public static validateLength(resourceName: string, inputName: string, minLength: number, maxLength: number, inputString?: string): void {
+    if (inputString !== undefined && (inputString.length < minLength || inputString.length > maxLength)) {
```

Comment:
One important note that I forgot to include in the Portfolio PR.

If the value provided is a dynamic CloudFormation expression, like a Parameter `Ref`, we should skip this validation. You can figure out whether a given value is a dynamic expression using the `Token.isUnresolved()` method.

This needs to be verified with a unit test.

Could you also retroactively change this in the Portfolio class?

Reply:
got it.  Followed convention from S3 test to make sure it doesn't throw validation error even when given an invalid value as token, which I think nicely covers the functionality. 

Will make quick PR for adding this to our portfolio construct in `aws-servicecatalog`

Reply:
btw, would there be any interest/support in making some input validation apart of `core`?

Reply:
created PR for this in servicecatalog: https://github.com/aws/aws-cdk/pull/15208